src/template.rs

```diff
@@ -147,8 +159,38 @@ impl<'a> Renderer<'a> {
             self.handlebars
                 .register_templates_directory(".hbs", templates)?;
         }
-        self.handlebars
-            .register_templates_directory(".hbs", &self.template_dir)?;
+        for entry in WalkDir::new(&self.template_dir)
+            .into_iter()
+            .filter_map(|e| e.ok())
+        {
+            if entry.file_type().is_file() && entry.path().extension().map_or(false, |e| e == "hbs")
```

Comment:
Again this chunk might be a good candidate for giving a name to because the intent is not obvious.

Reply:
chunked a part of it.